eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxPublisher.scala

```diff
@@ -73,20 +74,25 @@ object ReplaceableTxPublisher {
     }
   }
 
-  def getFeerate(feeEstimator: FeeEstimator, confirmBefore: BlockHeight, currentBlockHeight: BlockHeight): FeeratePerKw = {
-    val remainingBlocks = confirmBefore - currentBlockHeight
-    val blockTarget = remainingBlocks match {
-      // If our target is still very far in the future, no need to rush
-      case t if t >= 144 => 144
-      case t if t >= 72 => 72
-      case t if t >= 36 => 36
-      // However, if we get closer to the target, we start being more aggressive
-      case t if t >= 18 => 12
-      case t if t >= 12 => 6
-      case t if t >= 2 => 2
-      case _ => 1
+  def getFeerate(feeEstimator: FeeEstimator, confirmBefore: BlockHeight, currentBlockHeight: BlockHeight, hasEnoughSafeUtxos: Boolean): FeeratePerKw = {
+    if (hasEnoughSafeUtxos) {
+      val remainingBlocks = confirmBefore - currentBlockHeight
+      val blockTarget = remainingBlocks match {
+        // If our target is still very far in the future, no need to rush
+        case t if t >= 144 => 144
+        case t if t >= 72 => 72
+        case t if t >= 36 => 36
+        // However, if we get closer to the target, we start being more aggressive
+        case t if t >= 18 => 12
+        case t if t >= 12 => 6
+        case t if t >= 2 => 2
+        case _ => 1
+      }
+      feeEstimator.getFeeratePerKw(blockTarget)
+    } else {
+      // We don't have many safe utxos so we want the transaction to confirm quickly.
+      feeEstimator.getFeeratePerKw(1)
```

Comment:
I know it makes sense to target one block here, but I'm afraid we overpay. How about two blocks?

Reply:
Let's do 2 blocks, unless the confirmation target is in less than 6 blocks, in which case we do 1 block, ok?

Reply:
Done in https://github.com/ACINQ/eclair/pull/2141/commits/901b565fd492b81ea47400a9a13ed675e6e72566